index.js

```diff
@@ -20,7 +20,70 @@ mongoose
     return self.connection.dropDatabase();
   })
   .then(() => {
-    // Run your code here, after you have insured that the connection was made
+    //Iteration 1
+    // Recipe.create(data[0])
+    //   .then((res) => {
+    //     console.log("Data added", res);
+    //   })
+    //   .catch((err) => {
+    //     console.log("Something went wrong", err);
+    //   });
+
+
+    let insertRecipes = Recipe.insertMany(data);
+    // let updateRecipe;
+    // let deleteRecipe;
+
+    insertRecipes
+      .then((res) => {
+        console.log("Data added");
+
+        let updateRecipe = Recipe.findOneAndUpdate({ title: "Rigatoni alla Genovese" }, { $set: { duration: 100 } });
+        let deleteRecipe = Recipe.deleteOne({ title: "Carrot Cake" });
+        let disconnectPromise = Promise.all([updateRecipe, deleteRecipe]);
+
+        updateRecipe
+          .then((res) => {
+            console.log("Update success", res);
+          })
+          .catch((err) => {
+            console.log("Could not update", err);
+          });
+
+        deleteRecipe
+          .then((res) => {
+            console.log("Recipe deleted", res);
+          })
+          .catch((err) => {
+            console.log("Could not delete", err);
+          });
+
+        disconnectPromise
+          .then((res) => {
+            console.log(res);
+            mongoose.disconnect();
```

Comment:
disconnect() is itself a promise, so the console.log below will run even if there was a problem disconnecting. To solve this, wrap it inside a .then attached to the disconnect()